### PR TITLE
Researcher content

### PR DESCRIPTION
**Pull request by Walarks**
## About The Pull Request

Adds tools for researcher to collect research material and a green thing to "analyze" it with. To help with on-field operations, the green structure is draggable when unlocked, so the researcher can slowly drag it around with them to reposition it so it is easier to research material.

Research mechanics:
There are 4 item rarities - basic/common/uncommon/rare
Each resource has its own drop percentages for each rarity, which can be seen when the resource is inserted into the research computer. 
For example Tier 4 xeno research material has drop percentages 100/50/40/50. At basic rarity there are basic or common req points rewards, while at rare rarity there is a rare req points reward or cloak implant, and so on.
When researching an item, it can yield something from all 4 rarities if the roll is successful for each.

Research resource acquirement:
- xenomorph analyzer - click on a xeno(dead or alive) and get research resource(once per xeno) based on tier 1-4
- excavation tool - every 5 minutes up to 10 tiles from different areas groundside are marked and shown on the minimap for the researcher. If they can stand near a tile with an excavation mark and activate the excavation tool, they can excavate it to get a reward.

Research computer sprites by ArcaneMusic from tgstation/tgstation#48040

Possible expansion ideas:
- better TGUI
- ?add an additional roll for a rarity tier for every 100%?
- ?rewards other than money?
- ?adjust inserted resource percentages/drops based on additionally inserted resources?

**Barebones research UI with rewards info (changes colour based on inserted resource)**
Capture

**Basic preview:**
https://user-images.githubusercontent.com/45896482/137642548-ba55038b-5c11-4a4a-9b29-1e2681ec8e64.mp4

## Why It's Good For The Game
Researcher content that can help researcher be the frontline version of MO that it always wanted to be. Being or having a researcher can add more variety rounds since I don't think there is a role whole focus is collecting and working with various materials.

## Changelog
:cl:
add: researcher content
expansion: blood recovery implant(restores up to 200 blood), received from uncommon xeno research
/:cl:

### CODE REVIEW

**Comment by yyzsong** [comment]
IO moment

Could maybe add a hivemind message to warn xenos about digsites/current digging in progress (which they should totally give intel comps btw)

**Comment by AzonStarfall** [comment]
Money causes a system-breaking error when you sell it via the Requisitions lift. Takes down the whole system, permanently.

File: code/game/objects/machinery/computer/research.dm [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+/obj/machinery/computer/researchcomp/examine(user)
+	. = ..()
+	to_chat(user, span_notice(allowed_resources_desc))
+
+/obj/machinery/computer/researchcomp/update_icon()
+	..()
+	if(init_resource)
+		icon_state = "chamber_loaded"
+	else
```
[the remark this answers is not in the document]
Comment (Walarks) [reply code/game/objects/machinery/computer/research.dm]: moved researchcomp to inherit directly from machinery

File: code/game/objects/effects/landmarks/excavation_site_spawner.dm [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/obj/effect/landmark/excavation_site_spawner
```
Comment (BraveMole) [inline code/game/objects/effects/landmarks/excavation_site_spawner.dm:1]: i absolutly don't get why you would need a landmark spawner and a landmark site.

You could have done everything with just one landmark type

File: code/game/objects/effects/landmarks/excavation_site.dm [hunk 3]
```diff
@@ -1,5 +1,7 @@
 /obj/effect/landmark/excavation_site
 	name = "excavation landmark"
+	///Spawner the excavation site was spawned by
+	var/obj/effect/landmark/excavation_site_spawner/spawner
```
Comment (BraveMole) [inline code/game/objects/effects/landmarks/excavation_site.dm:4]: yeah you're making this too complex that it should.

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/landmarks/excavation_site_spawner.dm (hunk 2 above)
Comment (Walarks) [reply code/game/objects/effects/landmarks/excavation_site_spawner.dm:1]: changed to use only the spawner

File: code/game/objects/effects/landmarks/excavation_site.dm (hunk 3 above)
Comment (Walarks) [reply code/game/objects/effects/landmarks/excavation_site.dm]: ditto

**Review by BraveMole — APPROVED** [review]

**Pull request merged**